File: js-jasmine/src/items/specialitem.js
const agedBrie = require('./agedBrie');
const concertTickets = require('./concertTickets');
const Conjured = require('./conjured');
const Hand = require('./hand');

class specialItem{
    constructor(agedbrie = new agedBrie(), concerttickets = new concertTickets(), conjured = new Conjured(), hand = new Hand()){
        this.agedbrie = agedbrie
        this.concerttickets = concerttickets;
        this.conjured = conjured;
        this.hand = hand;
    }
    check(item){
        if(item.name === 'Aged Brie'){
            return this.agedbrie.update(item);
        }
        else if(item.name === 'Backstage passes to a TAFKAL80ETC concert'){
            return this.concerttickets.update(item);
        }
        else if(item.name === 'Conjured'){
            return this.conjured.update(item);
        }
        else{
            return this.hand.update(item);
        }
    }
}

module.exports = specialItem;

File: Java/src/main/java/com/gildedrose/items/SpecialItem.java
package com.gildedrose;

public class SpecialItem {
    private AgedBrie cheese = new AgedBrie();
    private Concert concert = new Concert();
    private Conjured conjured = new Conjured();

    public Item check(Item item){
        if(item.name == "Aged Brie"){
            return this.cheese.update(item);
        }else if(item.name == "Backstage passes to a TAFKAL80ETC concert"){
            return this.concert.update(item);
        }else if(item.name.contains("Conjured")){
            return this.conjured.update(item);
        }
        return item;
    }
}
